src/features/DebugSession.ts

```diff
@@ -386,3 +386,130 @@ export class PickPSHostProcessFeature implements IFeature {
+
+interface IGetRunspaceResponseBody {
+    runspaces: IRunspace[];
+}
+
+export class PickRunspaceFeature implements IFeature {
+
+    private command: vscode.Disposable;
+    private languageClient: LanguageClient;
+    private waitingForClientToken: vscode.CancellationTokenSource;
+    private getLanguageClientResolve: (value?: LanguageClient | Thenable<LanguageClient>) => void;
+
+    constructor() {
+
+        this.command =
+            vscode.commands.registerCommand("PowerShell.PickRunspace", () => {
+                return this.getLanguageClient()
+                           .then((_) => this.pickRunspace(), (_) => undefined);
+            });
+    }
+
+    public setLanguageClient(languageClient: LanguageClient) {
+        this.languageClient = languageClient;
+
+        if (this.waitingForClientToken) {
+            this.getLanguageClientResolve(this.languageClient);
+            this.clearWaitingToken();
+        }
+    }
+
+    public dispose() {
+        this.command.dispose();
+    }
+
+    private getLanguageClient(): Thenable<LanguageClient> {
+        if (this.languageClient) {
+            return Promise.resolve(this.languageClient);
+        } else {
+            // If PowerShell isn't finished loading yet, show a loading message
+            // until the LanguageClient is passed on to us
+            this.waitingForClientToken = new vscode.CancellationTokenSource();
+
+            return new Promise<LanguageClient>(
+                (resolve, reject) => {
+                    this.getLanguageClientResolve = resolve;
+
+                    vscode.window
+                        .showQuickPick(
+                            ["Cancel"],
+                            { placeHolder: "Attach to PowerShell host process: Please wait, starting PowerShell..." },
+                            this.waitingForClientToken.token)
+                        .then((response) => {
+                            if (response === "Cancel") {
+                                this.clearWaitingToken();
+                                reject();
+                            }
+                        });
+
+                    // Cancel the loading prompt after 60 seconds
+                    setTimeout(() => {
+                        if (this.waitingForClientToken) {
+                            this.clearWaitingToken();
+                            reject();
+
+                            vscode.window.showErrorMessage(
+                                "Attach to PowerShell host process: PowerShell session took too long to start.");
+                        }
+                    }, 60000);
+                },
+            );
+        }
+    }
+
+    private pickRunspace(): Thenable<string> {
+        return this.languageClient.sendRequest(GetRunspaceRequestType, null).then((runspaces) => {
+            const items: IRunspaceItem[] = [];
+
+            for (const p in runspaces) {
```

Comment:
you can probably switch this to a [for...of loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) then you don't have to index, you can just do:

```typescript
for (const runspace of runspaces)
```

and runspace is an `IRunspaceItem`

Reply:
Latest commit fixes this. There was actually a mismatch in the PSES response type and the VS Code response type.